src/index.js

```diff
@@ -11,6 +14,11 @@ export default function({ types: t }) {
         noParserImportDeclaration(path, state)
       },
       CallExpression(path, state) {
+        if(useUglifyPure(state)){
+          if(path.parent.type == 'VariableDeclarator' || path.parent.type == 'TaggedTemplateExpression'){
```

Comment:
this one seems to annotate **every** `CallExpression`, or Im not aware how this plugin gets activated - I havent studied the whole source, but from this piece alone it seems that it will annotate everything, ie. `myImpureGlobal()`

Reply:
Indeed yeah you will annotate everything but **once**.
As we were discussing before the issue of multiple pure comments were added, I fixed it by making sure that it is the outermost CallExpression only not the inner ones.

Reply:
If you are talking about that it would annotate undesired annotation so I haven't found this problem while testing, but if you have a specific test that I could test against to double check please give it to me.

Reply:
the point is that this plugin should only "touch" `styled-components` related code, because it is the only code about which we can make any kind of assumptions

Reply:
I get it now I will work on that.